src/NuGet.Core/NuGet.PackageManagement/IDE/PackageRestoreManager.cs

```diff
@@ -377,6 +378,15 @@ private NuGetPackageManager GetNuGetPackageManager(string solutionDirectory)
 
             packageRestoreContext.Token.ThrowIfCancellationRequested();
 
+            foreach (SourceRepository enabledSource in packageRestoreContext.SourceRepositories)
+            {
+                PackageSource source = enabledSource.PackageSource;
+                if (source.IsHttp && !source.IsHttps)
+                {
+                    packageRestoreContext.Logger.Log(LogLevel.Warning, string.Format(CultureInfo.CurrentCulture, Strings.Warning_HttpServerUsage, "restore", source.Source));
```

Comment:
I think we need to add log code to the warning.

Reply:
log codes only work in PackageReference.
For consistency we try not to add the log codes in packages.config.

Reply:
> For consistency we try not to add the log codes in packages.config as none of the resolver warnings/errors have any codes.

I don't understand this. With a log code, the customer has something to search for (or in VS, something to click in the Error List window), and then there can be a docs page where we can provide more detailed instructions on how to resolve the error.

Why is consistency with not using log codes matter more than this?

Reply:
You can't suppress warnings or elevate warnings in packages.config. Log codes usually suggest that you can do that. 
Also warnings/errors in packages.config and PackageReference are different because pc restore rarely puts any log messages there. 

I think the message itself should be actionable enough.